v3/src/components/graph/adornments/lsrl/lsrl-adornment-model.ts

```diff
@@ -0,0 +1,157 @@
+import { Instance, types } from "mobx-state-tree"
+import { Point } from "../../../data-display/data-display-types"
+import { AdornmentModel, IAdornmentModel, IUpdateCategoriesOptions, PointModel } from "../adornment-models"
+import { leastSquaresLinearRegression, tAt0975ForDf } from "../../utilities/graph-utils"
+import { kLSRLType } from "./lsrl-adornment-types"
+import { ICase } from "../../../../models/data/data-set-types"
+import { IGraphDataConfigurationModel } from "../../models/graph-data-configuration-model"
+import { ScaleNumericBaseType } from "../../../axis/axis-types"
+
+export const LSRLInstance = types.model("LSRLInstance", {
+  equationCoords: types.maybe(PointModel),
+  intercept: types.number,
+  rSquared: types.number,
+  slope: types.number,
+  sdResiduals: types.number
+})
+.actions(self => ({
+  setEquationCoords(coords: Point) {
+    self.equationCoords = PointModel.create(coords)
+  },
+  setIntercept(intercept: number) {
+    self.intercept = intercept
+  },
+  setRSquared(rSquared: number) {
+    self.rSquared = rSquared
+  },
+  setSlope(slope: number) {
+    self.slope = slope
+  },
+  setSdResiduals(sdResiduals: number) {
+    self.sdResiduals = sdResiduals
+  }
+}))
+
+export const LSRLAdornmentModel = AdornmentModel
+.named("LSRLAdornmentModel")
+.props({
+  type: types.optional(types.literal(kLSRLType), kLSRLType),
+  lines: types.map(types.array(LSRLInstance)),
+  showConfidenceBands: types.optional(types.boolean, false)
+})
+.views(() => ({
+  getCaseValues(
+    xAttrId: string, yAttrId: string, cellKey: Record<string, string>, dataConfig?: IGraphDataConfigurationModel,
+    cat?: string
+  ) {
+    if (!dataConfig) return []
+    const dataset = dataConfig?.dataset
+    const legendAttrId = dataConfig?.attributeID("legend")
+    const casesInPlot = dataConfig.subPlotCases(cellKey)
+    const caseValues: Point[] = []
+    casesInPlot.forEach((c: ICase) => {
+      const caseValueX = Number(dataset?.getValue(c.__id__, xAttrId))
+      const caseValueY = Number(dataset?.getValue(c.__id__, yAttrId))
```

Comment:
```suggestion
      const caseValueX = dataset?.getNumeric(c.__id__, xAttrId)
      const caseValueY = dataset?.getNumeric(c.__id__, yAttrId)
```
You don't need to perform the numeric conversion yourself.